app/controllers/criteria_controller.rb

```diff
@@ -51,14 +50,20 @@ def update
     flash.now[:success] = t('criterion_saved_success')
   end
 
-  # This method handles the drag/drop criteria sorting.
+  # Handles the drag/drop criteria sorting.
   def update_positions
     @assignment = Assignment.find(params[:assignment_id])
     @criteria = @assignment.get_criteria
 
     ActiveRecord::Base.transaction do
-      params[:criterion].
-        each_with_index { |id, index| @assignment.criterion_class.update(id, position: index + 1) if id != '' }
+      params[:criterion].each_with_index do |id, index|
+        begin
+          criterion_to_update = FlexibleCriterion.find(id)
```

Comment:
This isn't good (same comment as other pull request).

Need to modify the view to send not just id and index, but criterion type as well.